client/my-sites/sites/sites.jsx

```diff
@@ -93,4 +108,13 @@ export default React.createClass( {
 			</Main>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state ) => ( {
+		selectedSite: getSelectedSite( state ),
+		isPrivateSite: ( site ) => isPrivateSite( state, site.ID ),
```

Comment:
Those props which depend on `site` are used in the child component in here: https://github.com/Automattic/wp-calypso/blob/master/client/components/site-selector/index.jsx#L305. I think this filtering can be moved even further down the chain in here: https://github.com/Automattic/wp-calypso/blob/master/client/components/site-selector/index.jsx#L351. So whenever we are about to render one of the `<Site />` elements we could perform those checks. If it renders null, then we can mark such site as not visible.